### ISSUE TITLE
Power BI Custom Visual Not Working in Power BI Desktop

### ISSUE THREAD

**Issue by ABBVelocityDev** [microsoft/PowerBI-visuals#262]
I am using Power BI Desktop version 2.50.4859.502 64-bit (September, 2017).

We have a custom visual that uses ArcGIS for JavaScript, which in turn is built on the Dojo toolkit.  The custom visual works properly in Power BI Web using Chrome, IE, and Edge.  However, in Power BI desktop the visual does not work properly.  Adding code to the visual to debug, it appears that the Dojo require callback is never invoked.

I have attached a sample .pbix file, the custom visual, and Excel data that can be used to recreate a report using the visual as follows:
1. Import the data in the attached .xlsx file.
2. Add the custom visual.
3. Bind “Plant_ID” to the “Plant_ID” column.
4. Bind “Flowgate_ID” to the “Flowgate_ID” column.
5. Under the paint roller, set the “Security | API Key” field to: 3615d1e8-baef-47a4-aacc-7cd91894a423
6. Publish the report to Power BI Web and view it.

Thanks.
[Custom visual data.xlsx](https://github.com/Microsoft/PowerBI-visuals/files/1363106/Custom.visual.data.xlsx)
[evPowerPlantsAndConstraints.pbiviz.zip](https://github.com/Microsoft/PowerBI-visuals/files/1363108/evPowerPlantsAndConstraints.pbiviz.zip)
[Map Test.pbix.zip](https://github.com/Microsoft/PowerBI-visuals/files/1363109/Map.Test.pbix.zip)

**Comment by uve** [comment]
Hello @ABBVelocityDev, 
 Please try to use the following workaround:
https://github.com/Microsoft/PowerBI-visuals/issues/165

**Comment by ABBVelocityDev** [comment]
Hi.  Thanks for the suggestion; however, I don't fully understand how to try to implement the workaround.  

We dynamically load the ArcGIS and Dojo dependencies from js.arcgis.com, and do not have access to those scripts to make changes to the use of the window variable.  Is the suggested fix to download our own copies of the scripts and make changes?  Or is there another way to implement this?

Please advise.

Thanks.

**Comment by uve** [comment]
Hello @ABBVelocityDev,
 New powerbi-visuals-tools@1.8.0 has been released recently, please try to upgrade and test it again. 
External libs should work without any tricks.

**Comment by ABBVelocityDev** [comment]
I created a new custom visual using Power BI tools 1.8.0 and the Esri ArcGIS for JavaScript library.  Here's what I found:
* Using Power BI tools 1.8.0 and ArcGIS 4.1, the map appears in Power BI desktop, but the map tiles are rendered incorrectly - wrong tiles, wrong locations, and some tiles not rendered.
* Using Power BI tools 1.8.0 and ArcGIS 4.4, a simple map appears correctly in Power BI desktop.
* Using Power BI tools 1.3 and ArcGIS 4.4, The map appears in Power BI desktop and Power BI for the web, but issues outside the scope of this issue resurface (hit-testing and popups are not working correctly in either environment - but it seems consistent between Power BI desktop and web).

This issue seems to be as much related to ArcGIS for JavaScript as it is Power BI.

I'm not sure where to go from here.  Perhaps this can be closed.
